magicparse/builders.py

```diff
@@ -0,0 +1,92 @@
+from abc import ABC
+from decimal import Decimal
+
+from .transform import Transform
+
+
+class Builder(Transform, ABC):
+    @classmethod
+    def build(cls, options: dict) -> "Builder":
+        try:
+            name = options["name"]
+        except:
+            raise ValueError("builder must have a 'name' key")
+
+        try:
+            builder = cls.registry[name]
+        except:
+            raise ValueError(f"invalid builder '{name}'")
+
+        if "parameters" in options:
+            return builder(**options["parameters"])
+        else:
+            return builder()
+
+
+class Concat(Builder):
+    def __init__(self, fields: list[str]) -> None:
+        if (
+            not fields
+            or isinstance(fields, str)
```

Comment:
Si on inverse la condition ça peut être plus clair ?
all(isinstance(field, str) for field in fields) and len(fields) >= 2

Reply:
Pas possible :/ si la personne à la bonne idée de mettre un string...